src/core/exception.d

```diff
@@ -433,8 +433,15 @@ deprecated void setAssertHandler( AssertHandler h ) @trusted nothrow @nogc
  */
 extern (C) void onAssertError( string file = __FILE__, size_t line = __LINE__ ) nothrow
 {
-    if( _assertHandler is null )
-        throw new AssertError( file, line );
+    import core.stdc.string : memcpy;
+    char[1024] buffer;
```

Comment:
declare this buffer just within the if

1024 is random and a magic number, you could be concrete with char[__traits(classInstanceSize, AssertError)] buffer;